MetaMask/metamask-extension
# Fix #17884 - Remove Goerli buy link and disable button

## Explanation

Removes the ability to "buy" Goerli ETH to avoid redirect on old faucet.

* Fixes #17884

## Screenshots/Screencaps

<img width="633" alt="No Buy" src="https://user-images.githubusercontent.com/46655/224982798-db1917fb-ccce-4ab6-8b1d-83ef55955a4e.png">

## Manual Testing Steps

1.  Switch to Goerli network
2. See that the Buy link is disabled

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
